src/mono/mono/mini/mini-llvm.c

```diff
@@ -9594,6 +9582,18 @@ MONO_RESTORE_WARNING
 #endif
 
 #if defined(TARGET_X86) || defined(TARGET_AMD64) || defined(TARGET_WASM)
+		case OP_VECTOR_ANDN: {
+			LLVMValueRef minus_one [2];
+			minus_one [0] = const_int64 (-1);
+			minus_one [1] = const_int64 (-1);
+			LLVMValueRef vec_lhs_i64 = convert (ctx, lhs, sse_i8_t);
+			LLVMValueRef vec_xor = LLVMBuildXor (builder, vec_lhs_i64, LLVMConstVector (minus_one, 2), "");
+			LLVMValueRef vec_rhs_i64 = convert (ctx, rhs, sse_i8_t);
+			LLVMValueRef vec_and = LLVMBuildAnd (builder, vec_rhs_i64, vec_xor, "");
```

Comment:
NOTE: there is a difference between SSE_ANDN and common Vector128.AndNot that I also hit recently - https://github.com/dotnet/runtime/issues/77942

Basically, SSE's AndNot is doing `~X & Y` while `Vector128.AndNot` does the right thing - `X & ~Y`. It seems like your code is emitting SSE's variant here

Reply:
That is handled here https://github.com/dotnet/runtime/pull/78133/files#diff-012b47a27abc5dbc9d2a0b6fdef4adec27fa207a7d2b53823a8bc80f6539f5afL1190-L1192